src/lib/services/access-service.ts

```diff
@@ -578,10 +594,11 @@ export class AccessService {
         await this.validateRoleIsNotBuiltIn(id);
 
         const roleUsers = await this.getUsersForRole(id);
+        const roleGroups = await this.getGroupsForRole(id);
 
-        if (roleUsers.length > 0) {
+        if (roleUsers.length > 0 || roleGroups.length > 0) {
             throw new RoleInUseError(
-                'Role is in use by more than one user. You cannot delete a role that is in use without first removing the role from the users.',
+                'Role is in use by one or more users or groups. You cannot delete a role that is in use without first removing the role from the users and groups.',
```

Comment:
two ifs here and different error messages would probably make this easier to debug for the user

Reply:
Or just a simple users(0) or groups(23)

Reply:
Thanks @gardleopard! good feedback! I like the simple solution as it gives you overview of both regardless of which was the trigger. This is also part of an improvement of the UI that gives you an overview of usage before you attempt deleting